modules/exploits/example_linux_priv_esc.rb

```diff
@@ -0,0 +1,196 @@
+  end
+
+  # Pull the exploit binary or file (.c typically) from our system
+  def exploit_data(file)
+    ::File.binread ::File.join(Msf::Config.data_directory, 'exploits', 'DOES_NOT_EXIST', file)
+  end
+
+  # If we're going to live compile on the system, check gcc is installed
+  def live_compile?
+    return false unless datastore['COMPILE'].eql?('Auto') || datastore['COMPILE'].eql?('True')
+
+    if has_gcc?
+      vprint_good 'gcc is installed'
+      return true
+    end
+
+    unless datastore['COMPILE'].eql? 'Auto'
+      fail_with Failure::BadConfig, 'gcc is not installed. Compiling will fail.'
+    end
+  end
+
+  def check
+    # Check the kernel version to see if its in a vulnerable range
+    release = kernel_release
+    if Gem::Version.new(release.split('-').first) > Gem::Version.new('4.14.11') ||
+       Gem::Version.new(release.split('-').first) < Gem::Version.new('4.0')
+      vprint_error "Kernel version #{release} is not vulnerable"
+      return CheckCode::Safe
+    end
+    vprint_good "Kernel version #{release} appears to be vulnerable"
+
+    # Check the app is installed and the version, debian based example
+    package = cmd_exec('dpkg -l example | grep \'^ii\'')
+    if package && package.include?('1:2015.3.14AR.1-1build1')
+      print_good("Vulnerable app version #{package} detected")
+      CheckCode::Appears
+    end
+    CheckCode::Safe
+  end
+
+  #
+  # The exploit method drops a payload file to the system, then either compiles and runs
+  # or just runs the exploit on the system.
+  #
+  def exploit
+    # First check the system is vulnerable, or the user wants to run regardless
+    unless check == CheckCode::Appears
+      unless datastore['ForceExploit']
+        fail_with Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.'
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    # Check if we're already root
+    if is_root?
+      unless datastore['ForceExploit']
+        fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override'
+      end
+    end
+
+    # Make sure we can write our exploit and payload to the remote system
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    # Upload exploit executable, writing to a random name so AV doesn't have too easy a job
+    executable_name = ".#{rand_text_alphanumeric rand(5..10)}"
+    executable_path = "#{base_dir}/#{executable_name}"
+    if live_compile?
+      vprint_status 'Live compiling exploit on system...'
+      upload_and_compile executable_path, strip_comments(exploit_data('example.c'))
+      rm_f "#{executable_path}.c"
+    else
+      vprint_status 'Dropping pre-compiled exploit on system...'
+      upload_and_chmodx executable_path, exploit_data('example')
+    end
+
+    # Upload payload executable
+    payload_path = "#{base_dir}/.#{rand_text_alphanumeric rand(5..10)}"
```

Comment:
This notation is no longer required, as `rand_text_alphanumeric` accepts a `range`.

```suggestion
    payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
```